lib/karafka/connection/listener.rb

```diff
@@ -7,15 +7,27 @@ module Connection
     # @note Listener itself does nothing with the message - it will return to the block
     #   a raw Kafka::FetchedMessage
     class Listener
-      attr_reader :consumer_group
-
       # @param consumer_group [Karafka::Routing::ConsumerGroup] consumer group that holds details
       #   on what topics and with what settings should we listen
       # @return [Karafka::Connection::Listener] listener instance
       def initialize(consumer_group)
         @consumer_group = consumer_group
+        # This goes into the initialization, because we want to run it once even when client
+        # crashes, we don't to rerun callbacks. Also, callbacks should not interrupt with the
+        # main fetch loop
```

Comment:
This goes into the initialization because we want to run it once even when the client crashes, we don't to rerun callbacks. Also, callbacks should not interrupt with the main fetch loop